unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/BaseAvatar.cs

```diff
@@ -51,10 +54,21 @@ public async UniTask FadeOut(MeshRenderer targetRenderer, bool playParticles, Ca
             if (avatarRevealerContainer == null) 
                 return;
             
-            cancellationToken.ThrowIfCancellationRequested();
-
+            disposeCts ??= new CancellationTokenSource();
+            CancellationToken linkedCt = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, disposeCts.Token).Token;
+            linkedCt.ThrowIfCancellationRequested();
+            
             avatarRevealer.AddTarget(targetRenderer);
-            await avatarRevealer.StartAvatarRevealAnimation(playParticles, cancellationToken);
+            await avatarRevealer.StartAvatarRevealAnimation(playParticles && !isInsideHideArea, linkedCt);
+            
+            disposeCts?.Dispose();
+            disposeCts = null;
+        }
+        
+        public void ToggleHideAvatarArea(bool entered)
+        {
+            if(entered) disposeCts?.Cancel();
```

Comment:
You should dispose the CTS here as well.

Also, if you are cancelling the animation you have to set the final state. Otherwise, if the avatar goes into a hidden area in the middle of the transition, it will stay there forever.